src/fstests/fstests.jinja2

```diff
@@ -63,7 +109,9 @@ class Job(BaseJob):
         if self._check_kvm_xfstest_env():
             if self._kernel_config(src_path) == 'pass':
                 if self._kernel_build(src_path) == 'pass':
-                    return self._run_smoke_tests()
-        else:
-            return 'fail'
+                    if self._run_smoke_tests() == 'pass':
+                        if self._get_xml_from_vm(src_path) == 'pass':
+                            with open(os.path.join(src_path, 'results.json'), 'w') as result_file:
+                                result_file.write(json.dumps(self._parse_results(src_path)))
```

Comment:
We could add results for the kernel config and kernel build steps, and then also add the actual fstests results.  We could have `fstests.kernel.config`, `fstests.kernel.build` etc. for the tests run directly in this template and then add the results parsed from the XML file.

Then we should also catch any exception while parsing the results and sent the other results we already have, for example `fstests.smoke.run`, `fstests.smoke.xml`, `fstests.smoke.parse` for each method called earlier.

Reply:
To separate the results added by KernelCI from the actual fstests, we could have `fstests.kernelci.kernel.config`, `fstests.kernelci.smoke.run` etc. then `fstests.ext4.123` for the actual fstest results.  That way all the things added by KernelCI are in a separate sub-group called `kernelci` to make it clearer that they're not from the test suite.

Reply:
Ok, I will include these changes ahead, the submit parse seems to be a good PR for this refactor.